clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1284,6 +1292,11 @@ public ClientResponse disconnected(long timeMs, AuthenticationException authenti
                     true, null, authenticationException, null);
         }
 
+        public ClientResponse timedOut(long timeMs, AuthenticationException authenticationException) {
+            return new ClientResponse(header, callback, destination, createdTimeMs, timeMs,
+                    true, true, null, authenticationException, null);
```

Comment:
My understanding is that a timed out request is still considered disconnected as well. Am I correct? I think that this is important in order to not break compatibility. The NetworkClient is used in many other places.

Reply:
Correct. I have added a check in the `ClientResponse` constructor to ensure that time outs are only set when also disconnected.